### PR TITLE
3.0: Add image description attribute automatically

### PR DESCRIPTION
**Pull request by lukeseawalker**
Image description is added automatically at the end of the build, reporting info on the packages installed

Signed-off-by: Luca Carrogu <carrogu@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml [hunk 1]
```diff
@@ -58,27 +58,50 @@ phases:
             - |
               set -v
 
+              DESCRIPTION="AWS ParallelCluster AMI for {{ test.OperatingSystemName.outputs.stdout }}."
+              append_description () {
+                VALUE="$1"
+                if [[ ! "${VALUE}" =~ NOT_INSTALLED ]] && [[ -n "${VALUE}" ]]; then
```
Comment (yuleiwan) [inline cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml:64]: Minor: check the empty value first

File: cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml [hunk 2]
```diff
@@ -58,27 +58,50 @@ phases:
...
+                  echo "Appending ${VALUE} to decription"
+                  DESCRIPTION="${DESCRIPTION} ${VALUE}"
```
Comment (yuleiwan) [inline cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml:66]: a comma may be clearer than space to separate the versions

**Review by yuleiwan — APPROVED** [review]

File: cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml (hunk 1 above)
Comment (lukeseawalker) [reply cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml]: done

File: cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml (hunk 2 above)
Comment (lukeseawalker) [reply cli/src/pcluster/resources/imagebuilder/parallelcluster_tag.yaml]: done

**Pull request merged**